### ISSUE TITLE
Question of supervised train loss

### ISSUE THREAD

**Issue by Chris-WangQ** [kongds/Prompt-BERT#10]
Hello，
In the "model.py" , I finded the loss function of supervised training is CrossEntropyLoss. With hard negatives, why not select TripletMarginLoss? 
Looking forward to your reply, thanks!

**Comment by kongds** [comment]
hello, 

CrossEntropyLoss is implemented as contrastive learning (including hard negative) in supervised training following SimCSE.
The overall training objective is:
<img width="293" alt="image" src="https://user-images.githubusercontent.com/19282154/175486041-faf9e17b-179b-4238-b2ec-78d646472333.png">


And I think contrastive learning is more efficient than TripletMarginLoss as shown by SimCSE.
